{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pipeline to delimit ICE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### N.B. Charger l'enviro macsyfinder avant d'ouvrir ce notebook \n",
    "conda activate macsyfinder\n",
    "\n",
    "jupyter-notebook Tutorial_ICE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ce notebook de référence est tiré d'un article scientifique portant sur la détection des ICEs.\n",
    "\n",
    "Jean Cury, Marie Touchon, Eduardo P. C. Rocha, Integrative and conjugative elements and their hosts: composition, distribution and organization, Nucleic Acids Research, Volume 45, Issue 15, 6 September 2017, Pages 8943–8956, https://doi.org/10.1093/nar/gkx607\n",
    "\n",
    "Abstract\n",
    "\n",
    "Conjugation of single-stranded DNA drives horizontal gene transfer between bacteria and was widely studied in conjugative plasmids. The organization and function of integrative and conjugative elements (ICE), even if they are more abundant, was only studied in a few model systems. Comparative genomics of ICE has been precluded by the difficulty in finding and delimiting these elements. Here, we present the results of a method that circumvents these problems by requiring only the identification of the conjugation genes and the species’ pan-genome. We delimited 200 ICEs and this allowed the first large-scale characterization of these elements. We quantified the presence in ICEs of a wide set of functions associated with the biology of mobile genetic elements, including some that are typically associated with plasmids, such as partition and replication. Protein sequence similarity networks and phylogenetic analyses revealed that ICEs are structured in functional modules. Integrases and conjugation systems have different evolutionary histories, even if the gene repertoires of ICEs can be grouped in function of conjugation types. Our characterization of the composition and organization of ICEs paves the way for future functional and evolutionary analyses of their cargo genes, composed of a majority of unknown function genes.\n",
    "\n",
    "Dans le matériel supplémentaire de cet article (gkx607_Supp.zip), on peut trouver un tutoriel (jupyter-notebook) permettant de faire l'analyse des ICEs. Utilisé tel que, le tutoriel présente quelques bugs, qui empêche un utilisateur même expériementé de reproduire les résultats.\n",
    "\n",
    "J'ai adapté le code pour l'environnement particulier de la station de travail QCSHERA673892DX. En effet, l'annotation de prokka est particulière sur cette station de travail, car on souhaite annoter les gènes de résistance aux antibiotiques avec la dernière version de la base de données CARD.\n",
    "\n",
    "\n",
    "Jean-Simon Brouard le 23 décembre 2019\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Vérification des principales dépendances\n",
    "On devrait y voir roary, prokka, matplotlib, biopython, pandas, numpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# packages in environment at /home/brouardjs/miniconda3/envs/macsyfinder:\r\n",
      "biopython                 1.74             py27h516909a_0    conda-forge\r\n",
      "matplotlib                2.2.4                    py27_2    conda-forge\r\n",
      "matplotlib-base           2.2.4            py27h250f245_2    conda-forge\r\n",
      "numpy                     1.16.5           py27h95a1406_0    conda-forge\r\n",
      "pandas                    0.24.2           py27he6710b0_0    anaconda\r\n",
      "prokka                    1.14.5                  pl526_0    bioconda\r\n",
      "roary                     3.13.0          pl526h516909a_0    bioconda\r\n"
     ]
    }
   ],
   "source": [
    "!conda list | grep -e biopython -e roary -e matplotlib -e prokka -e macsyfinder -e pandas -e numpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####  Il faut aussi les exécutables usearch et le programme macsyfinder\n",
    "\n",
    "Note a moi-même, on voit qu'il y a deux dépendances qui n'ont pas ete installees dans l'environ macsyfinder, hum hum!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/brouardjs/miniconda3/envs/macsyfinder/bin/macsyfinder\n",
      "/home/brouardjs/bioinfo_programs/macsyfinder-1.0.5/usearch\n"
     ]
    }
   ],
   "source": [
    "!which macsyfinder\n",
    "!which usearch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/brouardjs/miniconda3/envs/macsyfinder/bin/prokka\r\n"
     ]
    }
   ],
   "source": [
    "#probleme de prokka\n",
    "\n",
    "!which prokka"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this notebook, we'll find ICE and delimit them in the Haemophilus influenzae species\n",
    "\n",
    "1. First we'll get the complete genome from NCBI\n",
    "3. We'll build the core genome\n",
    "2. We'll detect the conjugative system in the genomes\n",
    "4. We'll identify the core-genes flanking the conjugative system (defining a spot)\n",
    "5. We'll build the pan-genome of the spot\n",
    "6. We'll make a visual representation of the spot with gene family frequency to delimit the ICE."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p><div class=\"lev1\"><a href=\"#Data\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Data</a></div><div class=\"lev2\"><a href=\"#Download-fasta-sequences\"><span class=\"toc-item-num\">2.1&nbsp;&nbsp;</span>Download fasta sequences</a></div><div class=\"lev2\"><a href=\"#Make-CDS-annotation-with-Prokka\"><span class=\"toc-item-num\">2.2&nbsp;&nbsp;</span>Make CDS annotation with Prokka</a></div><div class=\"lev1\"><a href=\"#Build-Core-genomes\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Build Core genomes</a></div><div class=\"lev1\"><a href=\"#Identification-of-Conjugative-systems\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Identification of Conjugative systems</a></div><div class=\"lev1\"><a href=\"#Identify-the-flanking-core-genes:-finding-a-spot\"><span class=\"toc-item-num\">5&nbsp;&nbsp;</span>Identify the flanking core genes: finding a spot</a></div><div class=\"lev1\"><a href=\"#Pan-Genome-of-the-spot\"><span class=\"toc-item-num\">6&nbsp;&nbsp;</span>Pan-Genome of the spot</a></div><div class=\"lev1\"><a href=\"#Format-data-to-plot\"><span class=\"toc-item-num\">7&nbsp;&nbsp;</span>Format data to plot</a></div><div class=\"lev2\"><a href=\"#Make-a-table-for-each-ICE\"><span class=\"toc-item-num\">7.1&nbsp;&nbsp;</span>Make a table for each ICE</a></div><div class=\"lev1\"><a href=\"#Plot-the-spot\"><span class=\"toc-item-num\">8&nbsp;&nbsp;</span>Plot the spot</a></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Approximate time ot run this notebook: ~15-20 minutes.\n",
    "\n",
    "**NB**: The pipeline here works because the 3 ICEs we will found are in the same spot. But often, ICEs are found in different spots, and it implies to deal with that information to not mix everything."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Sequences/Replicon"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download fasta sequences"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One needs to download the data, let's to it with python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "Entrez.email = \"jean-simon.brouard@canada.ca\" #tells ncbi who you are."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I download the sequence from the NC number of 10 *Haemophilus influenzae*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "for nc in [\"NC_000907\", \"NC_007146\", \"NC_009566\", \"NC_009567\", \"NC_014920\", \"NC_014922\", \"NC_016809\", \"NC_017452\", \"NC_017451\", \"NC_022356\"]:\n",
    "    handle = Entrez.esearch(db=\"nucleotide\", term=nc, rettype=\"gb\", retmode=\"text\")\n",
    "    record = Entrez.read(handle)\n",
    "    if len(record[\"IdList\"]) == 0: # if no hit in genbank\n",
    "        print nc\n",
    "    handle2 = Entrez.efetch(db=\"nucleotide\", id=record[\"IdList\"][0], rettype=\"fasta\", retmode=\"text\")\n",
    "    replicons = SeqIO.read(handle2, \"fasta\")\n",
    "    SeqIO.write(replicons,\n",
    "                \"Sequences/Replicon/\" + nc +\".fst\",\n",
    "                \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   30504 Sequences/Replicon/NC_000907.fst\r\n",
      "   31910 Sequences/Replicon/NC_007146.fst\r\n",
      "   30219 Sequences/Replicon/NC_009566.fst\r\n",
      "   31455 Sequences/Replicon/NC_009567.fst\r\n",
      "   33099 Sequences/Replicon/NC_014920.fst\r\n",
      "   33452 Sequences/Replicon/NC_014922.fst\r\n",
      "   33027 Sequences/Replicon/NC_016809.fst\r\n",
      "   32207 Sequences/Replicon/NC_017451.fst\r\n",
      "   30324 Sequences/Replicon/NC_017452.fst\r\n",
      "   30938 Sequences/Replicon/NC_022356.fst\r\n",
      "  317135 total\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l Sequences/Replicon/*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make CDS annotation with Prokka"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Prokka](https://github.com/tseemann/prokka)\n",
    "\n",
    "\n",
    "#### Note importante a propos de l'annotation avec PROKKA - JSB\n",
    "\n",
    "Puisque nous avons une version modifiee de prokka qui utilise un db locale (et CARD), il est important de lancer l'annotation avec prokka dans l'environnement conda prokka en dehors de ce notebook!\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Sequences/Annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Donc aller dans un terminal et rouler en dehors de ce notebook\n",
    "\n",
    "conda activate prokka\n",
    "\n",
    "./prokka_DPL_laucher.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# A titre informatif, voici le contenu de mon script bash\n",
    "\n",
    "'''\n",
    "\n",
    "#!/bin/bash\n",
    "i=1\n",
    "for nc in \"NC_000907\" \"NC_007146\" \"NC_009566\" \"NC_009567\" \"NC_014920\" \"NC_014922\" \"NC_016809\" \"NC_017452\" \"NC_017451\" \"NC_022356\"; do\n",
    "    printf -v j \"%03d\" $i\n",
    "    echo HAIN\"$j\"\n",
    "    prokka --kingdom Bacteria \\\n",
    "            --genus Haemophilus Sequences/Replicon/\"$nc\".fst \\\n",
    "            --outdir Sequences/Annotations/HAIN\"$j\" \\\n",
    "            --proteins /data/ext4/dataDP/similarity_searches/db/card_original/protein_fasta_protein_homolog_model.fasta \\\n",
    "            --force \\\n",
    "            --locustag HAIN\"$j\" \\\n",
    "            --prefix HAIN\"$j\" \\\n",
    "            --cpus 0 >> LOG_prokka 2>&1\n",
    "    i=$((i+1))\n",
    "done\n",
    "\n",
    "\n",
    "'''"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build Core genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With [Roary](https://sanger-pathogens.github.io/Roary/)\n",
    "\n",
    "\n",
    "\n",
    "Move all `.gff` files in the same folder so Roary can work with them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Sequences/GFF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cp Sequences/Annotations/*/*gff Sequences/GFF/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Sequences/Pangenome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Pas de problème ici, roary est dans l'enviro macsyfinder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "!roary -p 20 -f Sequences/Pangenome/ Sequences/GFF/*.gff > LOG_roary 2>&1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get core-genes\n",
    "\n",
    "**BEWARE** to change the random number generated by `roary` (here `_1494846018`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "!query_pan_genome -g Sequences/Pangenome/_1576775051/clustered_proteins -a intersection Sequences/GFF/*.gff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "mv pan_genome_results core_genome.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gdhA: HAIN001_00203\tHAIN002_00273\tHAIN003_00473\tHAIN004_00707\tHAIN005_01972\tHAIN006_01111\tHAIN007_00257\tHAIN008_00454\tHAIN009_00399\tHAIN010_00871\r\n",
      "group_245: HAIN001_01230\tHAIN002_01260\tHAIN003_01252\tHAIN004_01896\tHAIN005_01094\tHAIN006_01885\tHAIN007_01365\tHAIN008_01198\tHAIN009_01245\tHAIN010_01455\r\n",
      "fbpA: HAIN001_00104\tHAIN002_00172\tHAIN003_00582\tHAIN004_00596\tHAIN005_00137\tHAIN006_00099\tHAIN007_00161\tHAIN008_00553\tHAIN009_00503\tHAIN010_01444\r\n",
      "group_502: HAIN001_00588\tHAIN002_00655\tHAIN003_00034\tHAIN004_01214\tHAIN005_01729\tHAIN006_00655\tHAIN007_00715\tHAIN008_00024\tHAIN009_00024\tHAIN010_00204\r\n",
      "rseB: HAIN001_00677\tHAIN002_00682\tHAIN003_01821\tHAIN004_01240\tHAIN005_01625\tHAIN006_00556\tHAIN007_00803\tHAIN008_01742\tHAIN009_01872\tHAIN010_00228\r\n",
      "menB: HAIN001_01035\tHAIN002_01096\tHAIN003_01426\tHAIN004_01693\tHAIN005_01259\tHAIN006_00161\tHAIN007_01162\tHAIN008_01363\tHAIN009_01421\tHAIN010_00602\r\n",
      "thrB: HAIN001_00094\tHAIN002_00163\tHAIN003_00589\tHAIN004_00590\tHAIN005_00131\tHAIN006_00089\tHAIN007_00152\tHAIN008_00560\tHAIN009_00512\tHAIN010_00848\r\n",
      "ribE: HAIN001_01367\tHAIN002_01517\tHAIN003_01047\tHAIN004_00279\tHAIN005_00812\tHAIN006_01478\tHAIN007_01505\tHAIN008_00995\tHAIN009_00886\tHAIN010_01070\r\n",
      "aspC: HAIN001_01689\tHAIN002_01345\tHAIN003_01158\tHAIN004_01994\tHAIN005_01007\tHAIN006_01800\tHAIN007_01849\tHAIN008_01110\tHAIN009_01160\tHAIN010_01276\r\n",
      "yqgF: HAIN001_00334\tHAIN002_00399\tHAIN003_00299\tHAIN004_00872\tHAIN005_00254\tHAIN006_00980\tHAIN007_00450\tHAIN008_00277\tHAIN009_00273\tHAIN010_01778\r\n"
     ]
    }
   ],
   "source": [
    "!head core_genome.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1084 core_genome.txt\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l core_genome.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove first columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "!awk -F\":\" '{print $2}' core_genome.txt > core_genome_2.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Identification of Conjugative systems"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I downloaded the [module CONJScan](https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/gem-pasteur/Macsyfinder_models/tree/master/Data/Conjugation) with model and profiles for conjugative systems to use with macsyfinder. I put it under a Conjugation folder and rename it \"Models\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Conjugation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls Conjugation/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# l'executable fonctionne\n",
    "!/usr/local/bin/macsyfinder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Pour reference, voici le contenu de l'excutable macsyfinder_laucher.sh\n",
    "'''\n",
    "#!/bin/bash\n",
    "for seq in Sequences/Annotations/*/*faa; do\n",
    "    name=${seq##*/}\n",
    "    for conj_type in typeF typeB typeC typeFATA typeFA typeG typeI typeT; do\n",
    "        macsyfinder \"$conj_type\" \\\n",
    "                    -w 20 \\\n",
    "                    --db-type ordered_replicon \\\n",
    "                    -d Conjugation/Models/DEF \\\n",
    "                    -p Conjugation/Models/HMM \\\n",
    "                    --profile-suffix .hmm \\\n",
    "                    --sequence-db \"$seq\" \\\n",
    "                    -o Conjugation/${name%%.*}\\_\"$conj_type\"\n",
    "    done\n",
    "done\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeFATA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN001_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN001/HAIN001.faa for system(s):\n",
      "\t- typeT\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeF\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeF  ---\n",
      "['HAIN002_00115', 'HAIN002_00117', 'HAIN002_00126', 'HAIN002_00159']\n",
      "['t4cp2', 'G_tfc7', 'virb4', 'MOBH']\n",
      "[111, 113, 122, 155]\n",
      "Considering typeF: \n",
      "-> uncomplete\n",
      "=> Putative typeF locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeB\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeB  ---\n",
      "['HAIN002_00115', 'HAIN002_00126', 'HAIN002_00159']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[111, 122, 155]\n",
      "Considering typeB: \n",
      "-> uncomplete\n",
      "=> Putative typeB locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeC\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeC  ---\n",
      "['HAIN002_00115', 'HAIN002_00126', 'HAIN002_00159']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[111, 122, 155]\n",
      "Considering typeC: \n",
      "-> uncomplete\n",
      "=> Putative typeC locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeFATA\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeFATA  ---\n",
      "['HAIN002_00115', 'HAIN002_00126', 'HAIN002_00159']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[111, 122, 155]\n",
      "Considering typeFATA: \n",
      "-> uncomplete\n",
      "=> Putative typeFATA locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeFA\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeFA  ---\n",
      "['HAIN002_00126', 'HAIN002_00159']\n",
      "['virb4', 'MOBH']\n",
      "[122, 155]\n",
      "Considering typeFA: \n",
      "-> uncomplete\n",
      "=> Putative typeFA locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeG\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeG  ---\n",
      "['HAIN002_00111', 'HAIN002_00112', 'HAIN002_00114', 'HAIN002_00115', 'HAIN002_00117', 'HAIN002_00118', 'HAIN002_00119', 'HAIN002_00120', 'HAIN002_00121', 'HAIN002_00122', 'HAIN002_00123', 'HAIN002_00124', 'HAIN002_00125', 'HAIN002_00126', 'HAIN002_00127', 'HAIN002_00128', 'HAIN002_00129', 'HAIN002_00136', 'HAIN002_00137', 'HAIN002_00138', 'HAIN002_00159']\n",
      "['G_tfc2', 'G_tfc3', 'G_tfc5', 't4cp2', 'G_tfc7', 'G_tfc8', 'G_tfc9', 'G_tfc10', 'G_tfc11', 'G_tfc12', 'G_tfc13', 'G_tfc14', 'G_tfc15', 'virb4', 'G_tfc17', 'G_tfc18', 'G_tfc19', 'G_tfc22', 'G_tfc23', 'G_tfc24', 'MOBH']\n",
      "[107, 108, 110, 111, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 132, 133, 134, 155]\n",
      "Considering typeG: \n",
      "-> single_locus\n",
      "====> Decision rule for putative system typeG:\n",
      "Mandatory genes: \n",
      "virb4\t1\n",
      "MOBB\t1\n",
      "t4cp1\t1\n",
      "Accessory genes: \n",
      "G_tfc17\t1\n",
      "G_tfc15\t1\n",
      "G_tfc14\t1\n",
      "G_tfc13\t1\n",
      "G_tfc12\t1\n",
      "G_tfc11\t1\n",
      "G_tfc10\t1\n",
      "G_tfc24\t1\n",
      "G_tfc22\t1\n",
      "G_tfc23\t1\n",
      "G_tfc19\t1\n",
      "G_tfc18\t1\n",
      "G_tfc7\t1\n",
      "G_tfc5\t1\n",
      "G_tfc3\t1\n",
      "G_tfc2\t1\n",
      "T_virB1\t0\n",
      "I_traE\t0\n",
      "G_tfc9\t1\n",
      "G_tfc8\t1\n",
      "\n",
      "nb_forbid : 0\n",
      "nb_mandat : 3\n",
      "nb_accessory : 18\n",
      "Complete \"single_locus\" system.\n",
      "******************************************\n",
      "\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter({'typeG_single_locus': 1})\n",
      "******************************************\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeI\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeI  ---\n",
      "['HAIN002_00115', 'HAIN002_00126', 'HAIN002_00159']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[111, 122, 155]\n",
      "Considering typeI: \n",
      "-> uncomplete\n",
      "=> Putative typeI locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN002_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN002/HAIN002.faa for system(s):\n",
      "\t- typeT\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeT  ---\n",
      "['HAIN002_00115', 'HAIN002_00126', 'HAIN002_00159']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[111, 122, 155]\n",
      "Considering typeT: \n",
      "-> uncomplete\n",
      "=> Putative typeT locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeFATA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN003_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN003/HAIN003.faa for system(s):\n",
      "\t- typeT\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeFATA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN004_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN004/HAIN004.faa for system(s):\n",
      "\t- typeT\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeFATA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN005_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN005/HAIN005.faa for system(s):\n",
      "\t- typeT\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeFATA\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN006_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN006/HAIN006.faa for system(s):\n",
      "\t- typeT\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeF\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeF  ---\n",
      "['HAIN007_00104', 'HAIN007_00106', 'HAIN007_00115', 'HAIN007_00148']\n",
      "['t4cp2', 'G_tfc7', 'virb4', 'MOBH']\n",
      "[100, 102, 111, 144]\n",
      "Considering typeF: \n",
      "-> uncomplete\n",
      "=> Putative typeF locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeB\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeB  ---\n",
      "['HAIN007_00104', 'HAIN007_00115', 'HAIN007_00148']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[100, 111, 144]\n",
      "Considering typeB: \n",
      "-> uncomplete\n",
      "=> Putative typeB locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeC\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeC  ---\n",
      "['HAIN007_00104', 'HAIN007_00115', 'HAIN007_00148']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[100, 111, 144]\n",
      "Considering typeC: \n",
      "-> uncomplete\n",
      "=> Putative typeC locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeFATA\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeFATA  ---\n",
      "['HAIN007_00104', 'HAIN007_00115', 'HAIN007_00148']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[100, 111, 144]\n",
      "Considering typeFATA: \n",
      "-> uncomplete\n",
      "=> Putative typeFATA locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeFA\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeFA  ---\n",
      "['HAIN007_00115', 'HAIN007_00148']\n",
      "['virb4', 'MOBH']\n",
      "[111, 144]\n",
      "Considering typeFA: \n",
      "-> uncomplete\n",
      "=> Putative typeFA locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeG\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeG  ---\n",
      "['HAIN007_00100', 'HAIN007_00101', 'HAIN007_00103', 'HAIN007_00104', 'HAIN007_00106', 'HAIN007_00107', 'HAIN007_00108', 'HAIN007_00109', 'HAIN007_00110', 'HAIN007_00111', 'HAIN007_00112', 'HAIN007_00113', 'HAIN007_00114', 'HAIN007_00115', 'HAIN007_00116', 'HAIN007_00117', 'HAIN007_00118', 'HAIN007_00125', 'HAIN007_00126', 'HAIN007_00127', 'HAIN007_00148']\n",
      "['G_tfc2', 'G_tfc3', 'G_tfc5', 't4cp2', 'G_tfc7', 'G_tfc8', 'G_tfc9', 'G_tfc10', 'G_tfc11', 'G_tfc12', 'G_tfc13', 'G_tfc14', 'G_tfc15', 'virb4', 'G_tfc17', 'G_tfc18', 'G_tfc19', 'G_tfc22', 'G_tfc23', 'G_tfc24', 'MOBH']\n",
      "[96, 97, 99, 100, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 121, 122, 123, 144]\n",
      "Considering typeG: \n",
      "-> single_locus\n",
      "====> Decision rule for putative system typeG:\n",
      "Mandatory genes: \n",
      "virb4\t1\n",
      "MOBB\t1\n",
      "t4cp1\t1\n",
      "Accessory genes: \n",
      "G_tfc17\t1\n",
      "G_tfc15\t1\n",
      "G_tfc14\t1\n",
      "G_tfc13\t1\n",
      "G_tfc12\t1\n",
      "G_tfc11\t1\n",
      "G_tfc10\t1\n",
      "G_tfc24\t1\n",
      "G_tfc22\t1\n",
      "G_tfc23\t1\n",
      "G_tfc19\t1\n",
      "G_tfc18\t1\n",
      "G_tfc7\t1\n",
      "G_tfc5\t1\n",
      "G_tfc3\t1\n",
      "G_tfc2\t1\n",
      "T_virB1\t0\n",
      "I_traE\t0\n",
      "G_tfc9\t1\n",
      "G_tfc8\t1\n",
      "\n",
      "nb_forbid : 0\n",
      "nb_mandat : 3\n",
      "nb_accessory : 18\n",
      "Complete \"single_locus\" system.\n",
      "******************************************\n",
      "\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter({'typeG_single_locus': 1})\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeI\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeI  ---\n",
      "['HAIN007_00104', 'HAIN007_00115', 'HAIN007_00148']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[100, 111, 144]\n",
      "Considering typeI: \n",
      "-> uncomplete\n",
      "=> Putative typeI locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN007_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN007/HAIN007.faa for system(s):\n",
      "\t- typeT\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeT  ---\n",
      "['HAIN007_00104', 'HAIN007_00115', 'HAIN007_00148']\n",
      "['t4cp2', 'virb4', 'MOBH']\n",
      "[100, 111, 144]\n",
      "Considering typeT: \n",
      "-> uncomplete\n",
      "=> Putative typeT locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeFATA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN008_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN008/HAIN008.faa for system(s):\n",
      "\t- typeT\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeF\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeF  ---\n",
      "['HAIN009_00556', 'HAIN009_00582', 'HAIN009_00591', 'HAIN009_00593']\n",
      "['MOBH', 'virb4', 'G_tfc7', 't4cp2']\n",
      "[527, 553, 562, 564]\n",
      "Considering typeF: \n",
      "-> uncomplete\n",
      "=> Putative typeF locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeB\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeB  ---\n",
      "['HAIN009_00556', 'HAIN009_00582', 'HAIN009_00593']\n",
      "['MOBH', 'virb4', 't4cp2']\n",
      "[527, 553, 564]\n",
      "Considering typeB: \n",
      "-> uncomplete\n",
      "=> Putative typeB locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeC\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeC  ---\n",
      "['HAIN009_00556', 'HAIN009_00582', 'HAIN009_00593']\n",
      "['MOBH', 'virb4', 't4cp2']\n",
      "[527, 553, 564]\n",
      "Considering typeC: \n",
      "-> uncomplete\n",
      "=> Putative typeC locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeFATA\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeFATA  ---\n",
      "['HAIN009_00556', 'HAIN009_00582', 'HAIN009_00593']\n",
      "['MOBH', 'virb4', 't4cp2']\n",
      "[527, 553, 564]\n",
      "Considering typeFATA: \n",
      "-> uncomplete\n",
      "=> Putative typeFATA locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeFA\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeFA  ---\n",
      "['HAIN009_00556', 'HAIN009_00582']\n",
      "['MOBH', 'virb4']\n",
      "[527, 553]\n",
      "Considering typeFA: \n",
      "-> uncomplete\n",
      "=> Putative typeFA locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeG\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeG  ---\n",
      "['HAIN009_00556', 'HAIN009_00570', 'HAIN009_00571', 'HAIN009_00572', 'HAIN009_00579', 'HAIN009_00580', 'HAIN009_00581', 'HAIN009_00582', 'HAIN009_00583', 'HAIN009_00584', 'HAIN009_00585', 'HAIN009_00586', 'HAIN009_00587', 'HAIN009_00588', 'HAIN009_00589', 'HAIN009_00590', 'HAIN009_00591', 'HAIN009_00593', 'HAIN009_00594', 'HAIN009_00596', 'HAIN009_00597']\n",
      "['MOBH', 'G_tfc24', 'G_tfc23', 'G_tfc22', 'G_tfc19', 'G_tfc18', 'G_tfc17', 'virb4', 'G_tfc15', 'G_tfc14', 'G_tfc13', 'G_tfc12', 'G_tfc11', 'G_tfc10', 'G_tfc9', 'G_tfc8', 'G_tfc7', 't4cp2', 'G_tfc5', 'G_tfc3', 'G_tfc2']\n",
      "[527, 541, 542, 543, 550, 551, 552, 553, 554, 555, 556, 557, 558, 559, 560, 561, 562, 564, 565, 567, 568]\n",
      "Considering typeG: \n",
      "-> single_locus\n",
      "====> Decision rule for putative system typeG:\n",
      "Mandatory genes: \n",
      "virb4\t1\n",
      "MOBB\t1\n",
      "t4cp1\t1\n",
      "Accessory genes: \n",
      "G_tfc17\t1\n",
      "G_tfc15\t1\n",
      "G_tfc14\t1\n",
      "G_tfc13\t1\n",
      "G_tfc12\t1\n",
      "G_tfc11\t1\n",
      "G_tfc10\t1\n",
      "G_tfc24\t1\n",
      "G_tfc22\t1\n",
      "G_tfc23\t1\n",
      "G_tfc19\t1\n",
      "G_tfc18\t1\n",
      "G_tfc7\t1\n",
      "G_tfc5\t1\n",
      "G_tfc3\t1\n",
      "G_tfc2\t1\n",
      "T_virB1\t0\n",
      "I_traE\t0\n",
      "G_tfc9\t1\n",
      "G_tfc8\t1\n",
      "\n",
      "nb_forbid : 0\n",
      "nb_mandat : 3\n",
      "nb_accessory : 18\n",
      "Complete \"single_locus\" system.\n",
      "******************************************\n",
      "\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter({'typeG_single_locus': 1})\n",
      "******************************************\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeI\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeI  ---\n",
      "['HAIN009_00556', 'HAIN009_00582', 'HAIN009_00593']\n",
      "['MOBH', 'virb4', 't4cp2']\n",
      "[527, 553, 564]\n",
      "Considering typeI: \n",
      "-> uncomplete\n",
      "=> Putative typeI locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN009_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN009/HAIN009.faa for system(s):\n",
      "\t- typeT\n",
      "\n",
      "************************************\n",
      " Analyzing clusters \n",
      "************************************\n",
      "\n",
      "\n",
      "--- Cluster typeT  ---\n",
      "['HAIN009_00556', 'HAIN009_00582', 'HAIN009_00593']\n",
      "['MOBH', 'virb4', 't4cp2']\n",
      "[527, 553, 564]\n",
      "Considering typeT: \n",
      "-> uncomplete\n",
      "=> Putative typeT locus stored for later treatment of scattered systems.\n",
      "\n",
      "    \n",
      "****************************************************\n",
      "******* Report scattered/uncompleted systems *******\n",
      "****************************************************\n",
      "\n",
      "******************************************\n",
      "\n",
      "******************************************\n",
      "Building reports of detected systems\n",
      " \n",
      "Counter()\n",
      "******************************************\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeF\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeF\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeB\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeB\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeC\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeC\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeFATA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeFATA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeFA\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeFA\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeG\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeG\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeI\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeI\n",
      "MacSyFinder's results will be stored in Conjugation/HAIN010_typeT\n",
      "Analysis launched on /data/ext4/dataDP/dev/fix_macsynfider/Macsyfinder_models/models/Conjugation/Sequences/Annotations/HAIN010/HAIN010.faa for system(s):\n",
      "\t- typeT\n"
     ]
    }
   ],
   "source": [
    "!./macsyfinder_laucher.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    0     0     0 Conjugation/HAIN002_typeB/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN002_typeC/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN002_typeF/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN002_typeFA/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN002_typeFATA/macsyfinder.report\r\n",
      "   22   352  3355 Conjugation/HAIN002_typeG/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN002_typeI/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN002_typeT/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeB/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeC/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeF/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeFA/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeFATA/macsyfinder.report\r\n",
      "   22   352  3352 Conjugation/HAIN007_typeG/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeI/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN007_typeT/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeB/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeC/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeF/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeFA/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeFATA/macsyfinder.report\r\n",
      "   22   352  3355 Conjugation/HAIN009_typeG/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeI/macsyfinder.report\r\n",
      "    0     0     0 Conjugation/HAIN009_typeT/macsyfinder.report\r\n",
      "   66  1056 10062 total\r\n"
     ]
    }
   ],
   "source": [
    "!wc Conjugation/*/macsyfinder.report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have 3 ICEs of type G in strain 2, 7 and 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "!awk 'NR==1||FNR>1' Conjugation/*/macsyfinder.report > Conj_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Identify the flanking core genes: finding a spot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# deprecated # conj = pd.read_table(\"Conj_results\", sep='\\t')\n",
    "conj = pd.read_csv(\"Conj_results\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "core = pd.read_csv(\"core_genome_2.txt\", sep='\\t', header=None, names=[\"HAIN{:03d}\".format(i) for i in range(1,11)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Hit_Id</th>\n",
       "      <th>Replicon_name</th>\n",
       "      <th>Position</th>\n",
       "      <th>Sequence_length</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Reference_system</th>\n",
       "      <th>Predicted_system</th>\n",
       "      <th>System_Id</th>\n",
       "      <th>System_status</th>\n",
       "      <th>Gene_status</th>\n",
       "      <th>i-evalue</th>\n",
       "      <th>Score</th>\n",
       "      <th>Profile_coverage</th>\n",
       "      <th>Sequence_coverage</th>\n",
       "      <th>Begin_match</th>\n",
       "      <th>End_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HAIN002_00111</td>\n",
       "      <td>UserReplicon</td>\n",
       "      <td>107</td>\n",
       "      <td>209</td>\n",
       "      <td>G_tfc2</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>2.600000e-60</td>\n",
       "      <td>200.1</td>\n",
       "      <td>0.921</td>\n",
       "      <td>0.990</td>\n",
       "      <td>2</td>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HAIN002_00112</td>\n",
       "      <td>UserReplicon</td>\n",
       "      <td>108</td>\n",
       "      <td>246</td>\n",
       "      <td>G_tfc3</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>5.800000e-88</td>\n",
       "      <td>290.7</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.972</td>\n",
       "      <td>7</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HAIN002_00114</td>\n",
       "      <td>UserReplicon</td>\n",
       "      <td>110</td>\n",
       "      <td>168</td>\n",
       "      <td>G_tfc5</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>6.600000e-56</td>\n",
       "      <td>185.4</td>\n",
       "      <td>0.983</td>\n",
       "      <td>0.994</td>\n",
       "      <td>2</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HAIN002_00115</td>\n",
       "      <td>UserReplicon</td>\n",
       "      <td>111</td>\n",
       "      <td>749</td>\n",
       "      <td>t4cp2</td>\n",
       "      <td>CONJ</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>mandatory</td>\n",
       "      <td>6.000000e-35</td>\n",
       "      <td>117.3</td>\n",
       "      <td>0.991</td>\n",
       "      <td>0.330</td>\n",
       "      <td>404</td>\n",
       "      <td>650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HAIN002_00117</td>\n",
       "      <td>UserReplicon</td>\n",
       "      <td>113</td>\n",
       "      <td>228</td>\n",
       "      <td>G_tfc7</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>8.400000e-85</td>\n",
       "      <td>281.1</td>\n",
       "      <td>0.972</td>\n",
       "      <td>0.991</td>\n",
       "      <td>3</td>\n",
       "      <td>228</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         #Hit_Id Replicon_name  Position  Sequence_length    Gene  \\\n",
       "0  HAIN002_00111  UserReplicon       107              209  G_tfc2   \n",
       "1  HAIN002_00112  UserReplicon       108              246  G_tfc3   \n",
       "2  HAIN002_00114  UserReplicon       110              168  G_tfc5   \n",
       "3  HAIN002_00115  UserReplicon       111              749   t4cp2   \n",
       "4  HAIN002_00117  UserReplicon       113              228  G_tfc7   \n",
       "\n",
       "  Reference_system Predicted_system             System_Id System_status  \\\n",
       "0            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "1            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "2            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "3             CONJ    typeG          UserReplicon_typeG_1  single_locus   \n",
       "4            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "\n",
       "  Gene_status      i-evalue  Score  Profile_coverage  Sequence_coverage  \\\n",
       "0   accessory  2.600000e-60  200.1             0.921              0.990   \n",
       "1   accessory  5.800000e-88  290.7             0.975              0.972   \n",
       "2   accessory  6.600000e-56  185.4             0.983              0.994   \n",
       "3   mandatory  6.000000e-35  117.3             0.991              0.330   \n",
       "4   accessory  8.400000e-85  281.1             0.972              0.991   \n",
       "\n",
       "   Begin_match  End_match  \n",
       "0            2        208  \n",
       "1            7        245  \n",
       "2            2        168  \n",
       "3          404        650  \n",
       "4            3        228  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conj.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reformat Macsyfinder output, replace UserReplicon by the name of the replicon. (I may have missed an option)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Attention code fragile, va chercher les 7 premiers caracters de la colonne 1\n",
    "conj[\"Replicon_name\"] = conj[\"#Hit_Id\"].apply(lambda x: x[:7])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Hit_Id</th>\n",
       "      <th>Replicon_name</th>\n",
       "      <th>Position</th>\n",
       "      <th>Sequence_length</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Reference_system</th>\n",
       "      <th>Predicted_system</th>\n",
       "      <th>System_Id</th>\n",
       "      <th>System_status</th>\n",
       "      <th>Gene_status</th>\n",
       "      <th>i-evalue</th>\n",
       "      <th>Score</th>\n",
       "      <th>Profile_coverage</th>\n",
       "      <th>Sequence_coverage</th>\n",
       "      <th>Begin_match</th>\n",
       "      <th>End_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HAIN002_00111</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>107</td>\n",
       "      <td>209</td>\n",
       "      <td>G_tfc2</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>2.600000e-60</td>\n",
       "      <td>200.1</td>\n",
       "      <td>0.921</td>\n",
       "      <td>0.990</td>\n",
       "      <td>2</td>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HAIN002_00112</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>108</td>\n",
       "      <td>246</td>\n",
       "      <td>G_tfc3</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>5.800000e-88</td>\n",
       "      <td>290.7</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.972</td>\n",
       "      <td>7</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HAIN002_00114</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>110</td>\n",
       "      <td>168</td>\n",
       "      <td>G_tfc5</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>6.600000e-56</td>\n",
       "      <td>185.4</td>\n",
       "      <td>0.983</td>\n",
       "      <td>0.994</td>\n",
       "      <td>2</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HAIN002_00115</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>111</td>\n",
       "      <td>749</td>\n",
       "      <td>t4cp2</td>\n",
       "      <td>CONJ</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>mandatory</td>\n",
       "      <td>6.000000e-35</td>\n",
       "      <td>117.3</td>\n",
       "      <td>0.991</td>\n",
       "      <td>0.330</td>\n",
       "      <td>404</td>\n",
       "      <td>650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HAIN002_00117</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>113</td>\n",
       "      <td>228</td>\n",
       "      <td>G_tfc7</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>UserReplicon_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>8.400000e-85</td>\n",
       "      <td>281.1</td>\n",
       "      <td>0.972</td>\n",
       "      <td>0.991</td>\n",
       "      <td>3</td>\n",
       "      <td>228</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         #Hit_Id Replicon_name  Position  Sequence_length    Gene  \\\n",
       "0  HAIN002_00111       HAIN002       107              209  G_tfc2   \n",
       "1  HAIN002_00112       HAIN002       108              246  G_tfc3   \n",
       "2  HAIN002_00114       HAIN002       110              168  G_tfc5   \n",
       "3  HAIN002_00115       HAIN002       111              749   t4cp2   \n",
       "4  HAIN002_00117       HAIN002       113              228  G_tfc7   \n",
       "\n",
       "  Reference_system Predicted_system             System_Id System_status  \\\n",
       "0            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "1            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "2            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "3             CONJ    typeG          UserReplicon_typeG_1  single_locus   \n",
       "4            typeG    typeG          UserReplicon_typeG_1  single_locus   \n",
       "\n",
       "  Gene_status      i-evalue  Score  Profile_coverage  Sequence_coverage  \\\n",
       "0   accessory  2.600000e-60  200.1             0.921              0.990   \n",
       "1   accessory  5.800000e-88  290.7             0.975              0.972   \n",
       "2   accessory  6.600000e-56  185.4             0.983              0.994   \n",
       "3   mandatory  6.000000e-35  117.3             0.991              0.330   \n",
       "4   accessory  8.400000e-85  281.1             0.972              0.991   \n",
       "\n",
       "   Begin_match  End_match  \n",
       "0            2        208  \n",
       "1            7        245  \n",
       "2            2        168  \n",
       "3          404        650  \n",
       "4            3        228  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conj.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "conj.System_Id.replace(\"UserReplicon\", conj.Replicon_name, regex=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Hit_Id</th>\n",
       "      <th>Replicon_name</th>\n",
       "      <th>Position</th>\n",
       "      <th>Sequence_length</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Reference_system</th>\n",
       "      <th>Predicted_system</th>\n",
       "      <th>System_Id</th>\n",
       "      <th>System_status</th>\n",
       "      <th>Gene_status</th>\n",
       "      <th>i-evalue</th>\n",
       "      <th>Score</th>\n",
       "      <th>Profile_coverage</th>\n",
       "      <th>Sequence_coverage</th>\n",
       "      <th>Begin_match</th>\n",
       "      <th>End_match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HAIN002_00111</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>107</td>\n",
       "      <td>209</td>\n",
       "      <td>G_tfc2</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>HAIN002_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>2.600000e-60</td>\n",
       "      <td>200.1</td>\n",
       "      <td>0.921</td>\n",
       "      <td>0.990</td>\n",
       "      <td>2</td>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HAIN002_00112</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>108</td>\n",
       "      <td>246</td>\n",
       "      <td>G_tfc3</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>HAIN002_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>5.800000e-88</td>\n",
       "      <td>290.7</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.972</td>\n",
       "      <td>7</td>\n",
       "      <td>245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HAIN002_00114</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>110</td>\n",
       "      <td>168</td>\n",
       "      <td>G_tfc5</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>HAIN002_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>6.600000e-56</td>\n",
       "      <td>185.4</td>\n",
       "      <td>0.983</td>\n",
       "      <td>0.994</td>\n",
       "      <td>2</td>\n",
       "      <td>168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HAIN002_00115</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>111</td>\n",
       "      <td>749</td>\n",
       "      <td>t4cp2</td>\n",
       "      <td>CONJ</td>\n",
       "      <td>typeG</td>\n",
       "      <td>HAIN002_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>mandatory</td>\n",
       "      <td>6.000000e-35</td>\n",
       "      <td>117.3</td>\n",
       "      <td>0.991</td>\n",
       "      <td>0.330</td>\n",
       "      <td>404</td>\n",
       "      <td>650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HAIN002_00117</td>\n",
       "      <td>HAIN002</td>\n",
       "      <td>113</td>\n",
       "      <td>228</td>\n",
       "      <td>G_tfc7</td>\n",
       "      <td>typeG</td>\n",
       "      <td>typeG</td>\n",
       "      <td>HAIN002_typeG_1</td>\n",
       "      <td>single_locus</td>\n",
       "      <td>accessory</td>\n",
       "      <td>8.400000e-85</td>\n",
       "      <td>281.1</td>\n",
       "      <td>0.972</td>\n",
       "      <td>0.991</td>\n",
       "      <td>3</td>\n",
       "      <td>228</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         #Hit_Id Replicon_name  Position  Sequence_length    Gene  \\\n",
       "0  HAIN002_00111       HAIN002       107              209  G_tfc2   \n",
       "1  HAIN002_00112       HAIN002       108              246  G_tfc3   \n",
       "2  HAIN002_00114       HAIN002       110              168  G_tfc5   \n",
       "3  HAIN002_00115       HAIN002       111              749   t4cp2   \n",
       "4  HAIN002_00117       HAIN002       113              228  G_tfc7   \n",
       "\n",
       "  Reference_system Predicted_system        System_Id System_status  \\\n",
       "0            typeG    typeG          HAIN002_typeG_1  single_locus   \n",
       "1            typeG    typeG          HAIN002_typeG_1  single_locus   \n",
       "2            typeG    typeG          HAIN002_typeG_1  single_locus   \n",
       "3             CONJ    typeG          HAIN002_typeG_1  single_locus   \n",
       "4            typeG    typeG          HAIN002_typeG_1  single_locus   \n",
       "\n",
       "  Gene_status      i-evalue  Score  Profile_coverage  Sequence_coverage  \\\n",
       "0   accessory  2.600000e-60  200.1             0.921              0.990   \n",
       "1   accessory  5.800000e-88  290.7             0.975              0.972   \n",
       "2   accessory  6.600000e-56  185.4             0.983              0.994   \n",
       "3   mandatory  6.000000e-35  117.3             0.991              0.330   \n",
       "4   accessory  8.400000e-85  281.1             0.972              0.991   \n",
       "\n",
       "   Begin_match  End_match  \n",
       "0            2        208  \n",
       "1            7        245  \n",
       "2            2        168  \n",
       "3          404        650  \n",
       "4            3        228  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conj.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HAIN001</th>\n",
       "      <th>HAIN002</th>\n",
       "      <th>HAIN003</th>\n",
       "      <th>HAIN004</th>\n",
       "      <th>HAIN005</th>\n",
       "      <th>HAIN006</th>\n",
       "      <th>HAIN007</th>\n",
       "      <th>HAIN008</th>\n",
       "      <th>HAIN009</th>\n",
       "      <th>HAIN010</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HAIN001_00203</td>\n",
       "      <td>HAIN002_00273</td>\n",
       "      <td>HAIN003_00473</td>\n",
       "      <td>HAIN004_00707</td>\n",
       "      <td>HAIN005_01972</td>\n",
       "      <td>HAIN006_01111</td>\n",
       "      <td>HAIN007_00257</td>\n",
       "      <td>HAIN008_00454</td>\n",
       "      <td>HAIN009_00399</td>\n",
       "      <td>HAIN010_00871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HAIN001_01230</td>\n",
       "      <td>HAIN002_01260</td>\n",
       "      <td>HAIN003_01252</td>\n",
       "      <td>HAIN004_01896</td>\n",
       "      <td>HAIN005_01094</td>\n",
       "      <td>HAIN006_01885</td>\n",
       "      <td>HAIN007_01365</td>\n",
       "      <td>HAIN008_01198</td>\n",
       "      <td>HAIN009_01245</td>\n",
       "      <td>HAIN010_01455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HAIN001_00104</td>\n",
       "      <td>HAIN002_00172</td>\n",
       "      <td>HAIN003_00582</td>\n",
       "      <td>HAIN004_00596</td>\n",
       "      <td>HAIN005_00137</td>\n",
       "      <td>HAIN006_00099</td>\n",
       "      <td>HAIN007_00161</td>\n",
       "      <td>HAIN008_00553</td>\n",
       "      <td>HAIN009_00503</td>\n",
       "      <td>HAIN010_01444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HAIN001_00588</td>\n",
       "      <td>HAIN002_00655</td>\n",
       "      <td>HAIN003_00034</td>\n",
       "      <td>HAIN004_01214</td>\n",
       "      <td>HAIN005_01729</td>\n",
       "      <td>HAIN006_00655</td>\n",
       "      <td>HAIN007_00715</td>\n",
       "      <td>HAIN008_00024</td>\n",
       "      <td>HAIN009_00024</td>\n",
       "      <td>HAIN010_00204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HAIN001_00677</td>\n",
       "      <td>HAIN002_00682</td>\n",
       "      <td>HAIN003_01821</td>\n",
       "      <td>HAIN004_01240</td>\n",
       "      <td>HAIN005_01625</td>\n",
       "      <td>HAIN006_00556</td>\n",
       "      <td>HAIN007_00803</td>\n",
       "      <td>HAIN008_01742</td>\n",
       "      <td>HAIN009_01872</td>\n",
       "      <td>HAIN010_00228</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          HAIN001        HAIN002        HAIN003        HAIN004        HAIN005  \\\n",
       "0   HAIN001_00203  HAIN002_00273  HAIN003_00473  HAIN004_00707  HAIN005_01972   \n",
       "1   HAIN001_01230  HAIN002_01260  HAIN003_01252  HAIN004_01896  HAIN005_01094   \n",
       "2   HAIN001_00104  HAIN002_00172  HAIN003_00582  HAIN004_00596  HAIN005_00137   \n",
       "3   HAIN001_00588  HAIN002_00655  HAIN003_00034  HAIN004_01214  HAIN005_01729   \n",
       "4   HAIN001_00677  HAIN002_00682  HAIN003_01821  HAIN004_01240  HAIN005_01625   \n",
       "\n",
       "         HAIN006        HAIN007        HAIN008        HAIN009        HAIN010  \n",
       "0  HAIN006_01111  HAIN007_00257  HAIN008_00454  HAIN009_00399  HAIN010_00871  \n",
       "1  HAIN006_01885  HAIN007_01365  HAIN008_01198  HAIN009_01245  HAIN010_01455  \n",
       "2  HAIN006_00099  HAIN007_00161  HAIN008_00553  HAIN009_00503  HAIN010_01444  \n",
       "3  HAIN006_00655  HAIN007_00715  HAIN008_00024  HAIN009_00024  HAIN010_00204  \n",
       "4  HAIN006_00556  HAIN007_00803  HAIN008_01742  HAIN009_01872  HAIN010_00228  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "core.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To identify the flanking core genes, I sort the column in the table `core` corresponding to the genome where I have an ICE, and find the index where I would insert any genes of that ICE\n",
    "\n",
    "On va trier le tableau core genome en utilisant 3 colonnes (les replicons HAIN002, HAIN007 et HAIN009)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dealing with HAIN002_typeG_1 which is a <type 'tuple'>\n",
      "sorting rows according to HAIN002\n",
      "('spot:', array([50]))\n",
      "dealing with HAIN007_typeG_1 which is a <type 'tuple'>\n",
      "sorting rows according to HAIN007\n",
      "('spot:', array([50]))\n",
      "dealing with HAIN009_typeG_1 which is a <type 'tuple'>\n",
      "sorting rows according to HAIN009\n",
      "('spot:', array([345]))\n"
     ]
    }
   ],
   "source": [
    "flanking_core = {}\n",
    "for ice in conj.groupby(\"System_Id\"):\n",
    "    print 'dealing with', ice[0], 'which is a', type(ice)\n",
    "    data = ice[1] # since ice[0] is just the name of the system_Id i.e. HAIN002_typeG_1\n",
    "    replicon = data.Replicon_name.unique()[0]\n",
    "    \n",
    "    print 'sorting rows according to', replicon\n",
    "    # sort the column in the table core corresponding to the genome where I have an ICE\n",
    "    core.sort_values(replicon, inplace=True)\n",
    "    #print (spots[0])\n",
    "    \n",
    "    spots = np.unique([core[replicon].searchsorted(gene) for gene in data[\"#Hit_Id\"]])\n",
    "    print('spot:', spots)\n",
    "    assert len(spots)==1 # make sure that all genes of the conjugative system insert at the same spot\n",
    "    flanking_core[ice[0]] = core.iloc[spots[0]-1:spots[0]+1].index\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HAIN002_typeG_1': Int64Index([267, 162], dtype='int64'),\n",
       " 'HAIN007_typeG_1': Int64Index([267, 162], dtype='int64'),\n",
       " 'HAIN009_typeG_1': Int64Index([162, 267], dtype='int64')}"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "flanking_core"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "core.sort_index(inplace=True) # reset the order of the table "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HAIN002_typeG_1': Int64Index([267, 162], dtype='int64'),\n",
       " 'HAIN007_typeG_1': Int64Index([267, 162], dtype='int64'),\n",
       " 'HAIN009_typeG_1': Int64Index([162, 267], dtype='int64')}"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "flanking_core"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "flanking_core_ID = core.loc[flanking_core.values()[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "N.B l'index du tableau est a 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HAIN001</th>\n",
       "      <th>HAIN002</th>\n",
       "      <th>HAIN003</th>\n",
       "      <th>HAIN004</th>\n",
       "      <th>HAIN005</th>\n",
       "      <th>HAIN006</th>\n",
       "      <th>HAIN007</th>\n",
       "      <th>HAIN008</th>\n",
       "      <th>HAIN009</th>\n",
       "      <th>HAIN010</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>HAIN001_00088</td>\n",
       "      <td>HAIN002_00089</td>\n",
       "      <td>HAIN003_00595</td>\n",
       "      <td>HAIN004_00583</td>\n",
       "      <td>HAIN005_00080</td>\n",
       "      <td>HAIN006_00083</td>\n",
       "      <td>HAIN007_00079</td>\n",
       "      <td>HAIN008_00566</td>\n",
       "      <td>HAIN009_00619</td>\n",
       "      <td>HAIN010_00854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>HAIN001_00093</td>\n",
       "      <td>HAIN002_00162</td>\n",
       "      <td>HAIN003_00590</td>\n",
       "      <td>HAIN004_00589</td>\n",
       "      <td>HAIN005_00130</td>\n",
       "      <td>HAIN006_00088</td>\n",
       "      <td>HAIN007_00151</td>\n",
       "      <td>HAIN008_00561</td>\n",
       "      <td>HAIN009_00513</td>\n",
       "      <td>HAIN010_00849</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            HAIN001        HAIN002        HAIN003        HAIN004  \\\n",
       "267   HAIN001_00088  HAIN002_00089  HAIN003_00595  HAIN004_00583   \n",
       "162   HAIN001_00093  HAIN002_00162  HAIN003_00590  HAIN004_00589   \n",
       "\n",
       "           HAIN005        HAIN006        HAIN007        HAIN008  \\\n",
       "267  HAIN005_00080  HAIN006_00083  HAIN007_00079  HAIN008_00566   \n",
       "162  HAIN005_00130  HAIN006_00088  HAIN007_00151  HAIN008_00561   \n",
       "\n",
       "           HAIN009        HAIN010  \n",
       "267  HAIN009_00619  HAIN010_00854  \n",
       "162  HAIN009_00513  HAIN010_00849  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "flanking_core_ID"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We see that they are all in the same spot, inbetween core genes 221 and 783, and that strain number 9 has been sequenced on the complementary strand or has a synteny breakpoint at that position. Let's eliminate that latter possibility.\n",
    "\n",
    "There is a breakpoint if the two flanking coregenes in the list of core genes are not continguous (in case of a inversion of one of the two coregenes with a third one for instance). Meaning that the difference between their rank is either 1, -1 or the number of coregenes (-1) (to wrap around).   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Getting their rank"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "order = core.values.argsort(axis=0)\n",
    "ranks = order.argsort(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "code_folding": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The ICE HAIN007_typeG_1 is in the spot flanked by coregenes 267 and 162 and there is no breakpoint in any strain\n",
      "The ICE HAIN009_typeG_1 is in the spot flanked by coregenes 162 and 267 and there is no breakpoint in any strain\n",
      "The ICE HAIN002_typeG_1 is in the spot flanked by coregenes 267 and 162 and there is no breakpoint in any strain\n"
     ]
    }
   ],
   "source": [
    "for k,v in flanking_core.iteritems():\n",
    "    \n",
    "    rank_diff = np.abs(np.diff(ranks[[v[0], v[1]]], axis=0)[0])\n",
    "    bkp = np.where((rank_diff!=1) & (rank_diff!=len(core)-1))[0]\n",
    "    if len(bkp)>0:\n",
    "        end_sentence = \"is breakpoint for strain {}\".format(bkp)\n",
    "    else:\n",
    "        end_sentence = \"is no breakpoint in any strain\"\n",
    "    print \"The ICE {} is in the spot flanked by coregenes {} and {} and there {}\".format(k, v[0], v[1], end_sentence)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here there is no breakpoint in any the strains at that spot. In case there is a breakpoint, one should remove the corresponding strain of the analysis. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's get the information on the relative sequencing orientation so we can normalize the strand position of the genes. We fix the reference with respect to the strain 2 (arbitrary, the first strain with an ICE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "sequencing_orientation = {i:j for i,j in zip(core.columns, np.median(np.diff(ranks[ranks[:,1].argsort()], axis=0), axis=0))}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'HAIN001': 1.0,\n",
       " 'HAIN002': 1.0,\n",
       " 'HAIN003': -1.0,\n",
       " 'HAIN004': 1.0,\n",
       " 'HAIN005': -1.0,\n",
       " 'HAIN006': -1.0,\n",
       " 'HAIN007': 1.0,\n",
       " 'HAIN008': -1.0,\n",
       " 'HAIN009': -1.0,\n",
       " 'HAIN010': 1.0}"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequencing_orientation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So the first two strain and the 7th and 10th are in the same median orientation, other strain will need to be flipped. We took the median orientation so it's not garanteed that it will work, as the ICE might be in a bigger region that have been inversed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pan-Genome of the spot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We now have the 2 flanking core genes. We'll gather all the genes that are in the spot and cluster them with uclust to make gene families."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([162, 267])"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(flanking_core.values())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "metI: HAIN001_00088\tHAIN002_00089\tHAIN003_00595\tHAIN004_00583\tHAIN005_00080\tHAIN006_00083\tHAIN007_00079\tHAIN008_00566\tHAIN009_00619\tHAIN010_00854\n",
    "\n",
    "thrC: HAIN001_00093\tHAIN002_00162\tHAIN003_00590\tHAIN004_00589\tHAIN005_00130\tHAIN006_00088\tHAIN007_00151\tHAIN008_00561\tHAIN009_00513\tHAIN010_00849\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HAIN001</th>\n",
       "      <th>HAIN002</th>\n",
       "      <th>HAIN003</th>\n",
       "      <th>HAIN004</th>\n",
       "      <th>HAIN005</th>\n",
       "      <th>HAIN006</th>\n",
       "      <th>HAIN007</th>\n",
       "      <th>HAIN008</th>\n",
       "      <th>HAIN009</th>\n",
       "      <th>HAIN010</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>HAIN001_00093</td>\n",
       "      <td>HAIN002_00162</td>\n",
       "      <td>HAIN003_00590</td>\n",
       "      <td>HAIN004_00589</td>\n",
       "      <td>HAIN005_00130</td>\n",
       "      <td>HAIN006_00088</td>\n",
       "      <td>HAIN007_00151</td>\n",
       "      <td>HAIN008_00561</td>\n",
       "      <td>HAIN009_00513</td>\n",
       "      <td>HAIN010_00849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>HAIN001_00088</td>\n",
       "      <td>HAIN002_00089</td>\n",
       "      <td>HAIN003_00595</td>\n",
       "      <td>HAIN004_00583</td>\n",
       "      <td>HAIN005_00080</td>\n",
       "      <td>HAIN006_00083</td>\n",
       "      <td>HAIN007_00079</td>\n",
       "      <td>HAIN008_00566</td>\n",
       "      <td>HAIN009_00619</td>\n",
       "      <td>HAIN010_00854</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            HAIN001        HAIN002        HAIN003        HAIN004  \\\n",
       "162   HAIN001_00093  HAIN002_00162  HAIN003_00590  HAIN004_00589   \n",
       "267   HAIN001_00088  HAIN002_00089  HAIN003_00595  HAIN004_00583   \n",
       "\n",
       "           HAIN005        HAIN006        HAIN007        HAIN008  \\\n",
       "162  HAIN005_00130  HAIN006_00088  HAIN007_00151  HAIN008_00561   \n",
       "267  HAIN005_00080  HAIN006_00083  HAIN007_00079  HAIN008_00566   \n",
       "\n",
       "           HAIN009        HAIN010  \n",
       "162  HAIN009_00513  HAIN010_00849  \n",
       "267  HAIN009_00619  HAIN010_00854  "
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "good_core = core.loc[np.unique(flanking_core.values())]\n",
    "good_core"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make list of gene for each interval "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Spots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = []\n",
    "for strain in good_core.columns:\n",
    "    strains.append(strain)\n",
    "    i = [v.strip() for v in good_core[strain].values]\n",
    "    f,l = int(i[0][-5:]), int(i[1][-5:])\n",
    "    s = SeqIO.index(\"Sequences/Annotations/{strain}/{strain}.faa\".format(strain=strain), \"fasta\")\n",
    "    seqs = [s.get(\"{}{:05d}\".format(i[0][:8], j)) for j in range(min(f,l), max(f,l)+1)]\n",
    "    SeqIO.write([seq for seq in seqs if seq is not None],\n",
    "                \"Spots/{}_spot.faa\".format(strain),\n",
    "                \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Here we only have 1 spot, but one should keep track of the spots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Spots/HAIN001_spot.faa:2\r\n",
      "Spots/HAIN002_spot.faa:70\r\n",
      "Spots/HAIN003_spot.faa:2\r\n",
      "Spots/HAIN004_spot.faa:3\r\n",
      "Spots/HAIN005_spot.faa:46\r\n",
      "Spots/HAIN006_spot.faa:2\r\n",
      "Spots/HAIN007_spot.faa:69\r\n",
      "Spots/HAIN008_spot.faa:2\r\n",
      "Spots/HAIN009_spot.faa:102\r\n",
      "Spots/HAIN010_spot.faa:2\r\n"
     ]
    }
   ],
   "source": [
    "!grep -c \">\" Spots/*faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat Spots/*faa > Spots/All_prot_spot.prt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir Spots/Clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "!usearch -quiet -cluster_fast Spots/All_prot_spot.prt \\\n",
    "         -id 0.7\\\n",
    "         -centroids Spots/All_prot_spot-70.fasta \\\n",
    "         -uc Spots/All_prot_spot-70.uc > Spots/log_usearch\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "cluster = pd.read_csv(\"Spots/All_prot_spot-70.uc\", sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00102 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN007_00092 hypothetical protein</td>\n",
       "      <td>HAIN002_00102 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>172</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00606 hypothetical protein</td>\n",
       "      <td>HAIN002_00102 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S</td>\n",
       "      <td>1</td>\n",
       "      <td>156</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00108 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H</td>\n",
       "      <td>1</td>\n",
       "      <td>156</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>156</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00600 hypothetical protein</td>\n",
       "      <td>HAIN002_00108 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S</td>\n",
       "      <td>2</td>\n",
       "      <td>198</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00109 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>H</td>\n",
       "      <td>2</td>\n",
       "      <td>198</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>198</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00599 hypothetical protein</td>\n",
       "      <td>HAIN002_00109 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>S</td>\n",
       "      <td>3</td>\n",
       "      <td>209</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00111 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>H</td>\n",
       "      <td>3</td>\n",
       "      <td>209</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00597 hypothetical protein</td>\n",
       "      <td>HAIN002_00111 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>S</td>\n",
       "      <td>4</td>\n",
       "      <td>168</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00114 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>H</td>\n",
       "      <td>4</td>\n",
       "      <td>168</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>168</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00594 hypothetical protein</td>\n",
       "      <td>HAIN002_00114 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>S</td>\n",
       "      <td>5</td>\n",
       "      <td>133</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00120 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>H</td>\n",
       "      <td>5</td>\n",
       "      <td>133</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN007_00109 hypothetical protein</td>\n",
       "      <td>HAIN002_00120 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>H</td>\n",
       "      <td>5</td>\n",
       "      <td>133</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>133</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00588 hypothetical protein</td>\n",
       "      <td>HAIN002_00120 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>S</td>\n",
       "      <td>6</td>\n",
       "      <td>135</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00125 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>H</td>\n",
       "      <td>6</td>\n",
       "      <td>135</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>135</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN007_00114 hypothetical protein</td>\n",
       "      <td>HAIN002_00125 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>H</td>\n",
       "      <td>6</td>\n",
       "      <td>135</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>135</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00583 hypothetical protein</td>\n",
       "      <td>HAIN002_00125 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>S</td>\n",
       "      <td>7</td>\n",
       "      <td>147</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00127 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>H</td>\n",
       "      <td>7</td>\n",
       "      <td>147</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>147</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN007_00116 hypothetical protein</td>\n",
       "      <td>HAIN002_00127 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>H</td>\n",
       "      <td>7</td>\n",
       "      <td>147</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>147</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00581 hypothetical protein</td>\n",
       "      <td>HAIN002_00127 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>S</td>\n",
       "      <td>8</td>\n",
       "      <td>59</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00134 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>59</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN007_00123 hypothetical protein</td>\n",
       "      <td>HAIN002_00134 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>H</td>\n",
       "      <td>8</td>\n",
       "      <td>59</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>59</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN009_00574 hypothetical protein</td>\n",
       "      <td>HAIN002_00134 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>S</td>\n",
       "      <td>9</td>\n",
       "      <td>317</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00143 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>H</td>\n",
       "      <td>9</td>\n",
       "      <td>317</td>\n",
       "      <td>100.0</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>317</td>\n",
       "      <td>=</td>\n",
       "      <td>HAIN007_00132 hypothetical protein</td>\n",
       "      <td>HAIN002_00143 hypothetical protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>S</td>\n",
       "      <td>10</td>\n",
       "      <td>92</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00090 Killer protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>S</td>\n",
       "      <td>11</td>\n",
       "      <td>102</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00097 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>S</td>\n",
       "      <td>12</td>\n",
       "      <td>150</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00106 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>S</td>\n",
       "      <td>13</td>\n",
       "      <td>86</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00524 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>S</td>\n",
       "      <td>14</td>\n",
       "      <td>607</td>\n",
       "      <td>*</td>\n",
       "      <td>.</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00530 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>C</td>\n",
       "      <td>103</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00107 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>C</td>\n",
       "      <td>104</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00122 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>C</td>\n",
       "      <td>105</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00133 [Salmonella enterica subsp. ente...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>C</td>\n",
       "      <td>106</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00107 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>C</td>\n",
       "      <td>107</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00122 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>C</td>\n",
       "      <td>108</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00124 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>C</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00518 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>C</td>\n",
       "      <td>110</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00557 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>C</td>\n",
       "      <td>111</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00121 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>C</td>\n",
       "      <td>112</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00098 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>C</td>\n",
       "      <td>113</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00118 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>C</td>\n",
       "      <td>114</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00125 Putative membrane protein insert...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>C</td>\n",
       "      <td>115</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00535 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>C</td>\n",
       "      <td>116</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00116 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>C</td>\n",
       "      <td>117</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00132 transposase</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>C</td>\n",
       "      <td>118</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00152 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>C</td>\n",
       "      <td>119</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00155 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>C</td>\n",
       "      <td>120</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN007_00099 IncF plasmid conjugative transfe...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>C</td>\n",
       "      <td>121</td>\n",
       "      <td>2</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00527 Hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>C</td>\n",
       "      <td>122</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00554 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>C</td>\n",
       "      <td>123</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN009_00563 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>C</td>\n",
       "      <td>124</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00096 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>C</td>\n",
       "      <td>125</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00099 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>C</td>\n",
       "      <td>126</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00104 DNA topoisomerase III</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>C</td>\n",
       "      <td>127</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00115 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>C</td>\n",
       "      <td>128</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00117 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>C</td>\n",
       "      <td>129</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00135 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>C</td>\n",
       "      <td>130</td>\n",
       "      <td>3</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN002_00151 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>C</td>\n",
       "      <td>131</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN004_00584 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>C</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>*</td>\n",
       "      <td>HAIN005_00113 hypothetical protein</td>\n",
       "      <td>*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>433 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     0    1    2      3  4  5    6  7  \\\n",
       "0    S    0  172      *  .  *    *  *   \n",
       "1    H    0  172  100.0  .  0  172  =   \n",
       "2    H    0  172  100.0  .  0  172  =   \n",
       "3    S    1  156      *  .  *    *  *   \n",
       "4    H    1  156  100.0  .  0  156  =   \n",
       "5    S    2  198      *  .  *    *  *   \n",
       "6    H    2  198  100.0  .  0  198  =   \n",
       "7    S    3  209      *  .  *    *  *   \n",
       "8    H    3  209  100.0  .  0  209  =   \n",
       "9    S    4  168      *  .  *    *  *   \n",
       "10   H    4  168  100.0  .  0  168  =   \n",
       "11   S    5  133      *  .  *    *  *   \n",
       "12   H    5  133  100.0  .  0  133  =   \n",
       "13   H    5  133  100.0  .  0  133  =   \n",
       "14   S    6  135      *  .  *    *  *   \n",
       "15   H    6  135  100.0  .  0  135  =   \n",
       "16   H    6  135  100.0  .  0  135  =   \n",
       "17   S    7  147      *  .  *    *  *   \n",
       "18   H    7  147  100.0  .  0  147  =   \n",
       "19   H    7  147  100.0  .  0  147  =   \n",
       "20   S    8   59      *  .  *    *  *   \n",
       "21   H    8   59  100.0  .  0   59  =   \n",
       "22   H    8   59  100.0  .  0   59  =   \n",
       "23   S    9  317      *  .  *    *  *   \n",
       "24   H    9  317  100.0  .  0  317  =   \n",
       "25   S   10   92      *  .  *    *  *   \n",
       "26   S   11  102      *  .  *    *  *   \n",
       "27   S   12  150      *  .  *    *  *   \n",
       "28   S   13   86      *  .  *    *  *   \n",
       "29   S   14  607      *  .  *    *  *   \n",
       "..  ..  ...  ...    ... .. ..  ... ..   \n",
       "403  C  103    2      *  *  *    *  *   \n",
       "404  C  104    3      *  *  *    *  *   \n",
       "405  C  105    3      *  *  *    *  *   \n",
       "406  C  106    2      *  *  *    *  *   \n",
       "407  C  107    1      *  *  *    *  *   \n",
       "408  C  108    1      *  *  *    *  *   \n",
       "409  C  109    1      *  *  *    *  *   \n",
       "410  C  110    1      *  *  *    *  *   \n",
       "411  C  111    3      *  *  *    *  *   \n",
       "412  C  112    2      *  *  *    *  *   \n",
       "413  C  113    2      *  *  *    *  *   \n",
       "414  C  114    1      *  *  *    *  *   \n",
       "415  C  115    1      *  *  *    *  *   \n",
       "416  C  116    3      *  *  *    *  *   \n",
       "417  C  117    3      *  *  *    *  *   \n",
       "418  C  118    2      *  *  *    *  *   \n",
       "419  C  119    2      *  *  *    *  *   \n",
       "420  C  120    3      *  *  *    *  *   \n",
       "421  C  121    2      *  *  *    *  *   \n",
       "422  C  122    1      *  *  *    *  *   \n",
       "423  C  123    1      *  *  *    *  *   \n",
       "424  C  124    3      *  *  *    *  *   \n",
       "425  C  125    3      *  *  *    *  *   \n",
       "426  C  126    3      *  *  *    *  *   \n",
       "427  C  127    3      *  *  *    *  *   \n",
       "428  C  128    3      *  *  *    *  *   \n",
       "429  C  129    3      *  *  *    *  *   \n",
       "430  C  130    3      *  *  *    *  *   \n",
       "431  C  131    1      *  *  *    *  *   \n",
       "432  C  132    1      *  *  *    *  *   \n",
       "\n",
       "                                                     8  \\\n",
       "0                   HAIN002_00102 hypothetical protein   \n",
       "1                   HAIN007_00092 hypothetical protein   \n",
       "2                   HAIN009_00606 hypothetical protein   \n",
       "3                   HAIN002_00108 hypothetical protein   \n",
       "4                   HAIN009_00600 hypothetical protein   \n",
       "5                   HAIN002_00109 hypothetical protein   \n",
       "6                   HAIN009_00599 hypothetical protein   \n",
       "7                   HAIN002_00111 hypothetical protein   \n",
       "8                   HAIN009_00597 hypothetical protein   \n",
       "9                   HAIN002_00114 hypothetical protein   \n",
       "10                  HAIN009_00594 hypothetical protein   \n",
       "11                  HAIN002_00120 hypothetical protein   \n",
       "12                  HAIN007_00109 hypothetical protein   \n",
       "13                  HAIN009_00588 hypothetical protein   \n",
       "14                  HAIN002_00125 hypothetical protein   \n",
       "15                  HAIN007_00114 hypothetical protein   \n",
       "16                  HAIN009_00583 hypothetical protein   \n",
       "17                  HAIN002_00127 hypothetical protein   \n",
       "18                  HAIN007_00116 hypothetical protein   \n",
       "19                  HAIN009_00581 hypothetical protein   \n",
       "20                  HAIN002_00134 hypothetical protein   \n",
       "21                  HAIN007_00123 hypothetical protein   \n",
       "22                  HAIN009_00574 hypothetical protein   \n",
       "23                  HAIN002_00143 hypothetical protein   \n",
       "24                  HAIN007_00132 hypothetical protein   \n",
       "25                        HAIN005_00090 Killer protein   \n",
       "26                  HAIN005_00097 hypothetical protein   \n",
       "27                  HAIN005_00106 hypothetical protein   \n",
       "28                  HAIN009_00524 hypothetical protein   \n",
       "29                  HAIN009_00530 hypothetical protein   \n",
       "..                                                 ...   \n",
       "403                 HAIN002_00107 hypothetical protein   \n",
       "404                 HAIN002_00122 hypothetical protein   \n",
       "405  HAIN002_00133 [Salmonella enterica subsp. ente...   \n",
       "406                 HAIN005_00107 hypothetical protein   \n",
       "407                 HAIN005_00122 hypothetical protein   \n",
       "408                 HAIN005_00124 hypothetical protein   \n",
       "409                 HAIN009_00518 hypothetical protein   \n",
       "410                 HAIN009_00557 hypothetical protein   \n",
       "411                 HAIN002_00121 hypothetical protein   \n",
       "412                 HAIN005_00098 hypothetical protein   \n",
       "413                 HAIN005_00118 hypothetical protein   \n",
       "414  HAIN005_00125 Putative membrane protein insert...   \n",
       "415                 HAIN009_00535 hypothetical protein   \n",
       "416                 HAIN002_00116 hypothetical protein   \n",
       "417                          HAIN002_00132 transposase   \n",
       "418                 HAIN002_00152 hypothetical protein   \n",
       "419                 HAIN002_00155 hypothetical protein   \n",
       "420  HAIN007_00099 IncF plasmid conjugative transfe...   \n",
       "421                 HAIN009_00527 Hypothetical protein   \n",
       "422                 HAIN009_00554 hypothetical protein   \n",
       "423                 HAIN009_00563 hypothetical protein   \n",
       "424                 HAIN002_00096 hypothetical protein   \n",
       "425                 HAIN002_00099 hypothetical protein   \n",
       "426                HAIN002_00104 DNA topoisomerase III   \n",
       "427                 HAIN002_00115 hypothetical protein   \n",
       "428                 HAIN002_00117 hypothetical protein   \n",
       "429                 HAIN002_00135 hypothetical protein   \n",
       "430                 HAIN002_00151 hypothetical protein   \n",
       "431  HAIN004_00584 Cystathionine gamma-synthase/O-a...   \n",
       "432                 HAIN005_00113 hypothetical protein   \n",
       "\n",
       "                                      9  \n",
       "0                                     *  \n",
       "1    HAIN002_00102 hypothetical protein  \n",
       "2    HAIN002_00102 hypothetical protein  \n",
       "3                                     *  \n",
       "4    HAIN002_00108 hypothetical protein  \n",
       "5                                     *  \n",
       "6    HAIN002_00109 hypothetical protein  \n",
       "7                                     *  \n",
       "8    HAIN002_00111 hypothetical protein  \n",
       "9                                     *  \n",
       "10   HAIN002_00114 hypothetical protein  \n",
       "11                                    *  \n",
       "12   HAIN002_00120 hypothetical protein  \n",
       "13   HAIN002_00120 hypothetical protein  \n",
       "14                                    *  \n",
       "15   HAIN002_00125 hypothetical protein  \n",
       "16   HAIN002_00125 hypothetical protein  \n",
       "17                                    *  \n",
       "18   HAIN002_00127 hypothetical protein  \n",
       "19   HAIN002_00127 hypothetical protein  \n",
       "20                                    *  \n",
       "21   HAIN002_00134 hypothetical protein  \n",
       "22   HAIN002_00134 hypothetical protein  \n",
       "23                                    *  \n",
       "24   HAIN002_00143 hypothetical protein  \n",
       "25                                    *  \n",
       "26                                    *  \n",
       "27                                    *  \n",
       "28                                    *  \n",
       "29                                    *  \n",
       "..                                  ...  \n",
       "403                                   *  \n",
       "404                                   *  \n",
       "405                                   *  \n",
       "406                                   *  \n",
       "407                                   *  \n",
       "408                                   *  \n",
       "409                                   *  \n",
       "410                                   *  \n",
       "411                                   *  \n",
       "412                                   *  \n",
       "413                                   *  \n",
       "414                                   *  \n",
       "415                                   *  \n",
       "416                                   *  \n",
       "417                                   *  \n",
       "418                                   *  \n",
       "419                                   *  \n",
       "420                                   *  \n",
       "421                                   *  \n",
       "422                                   *  \n",
       "423                                   *  \n",
       "424                                   *  \n",
       "425                                   *  \n",
       "426                                   *  \n",
       "427                                   *  \n",
       "428                                   *  \n",
       "429                                   *  \n",
       "430                                   *  \n",
       "431                                   *  \n",
       "432                                   *  \n",
       "\n",
       "[433 rows x 10 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "families = cluster[cluster[0]!=\"C\"].groupby(1)[8].apply(lambda x: sorted(x)).values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "fam = pd.DataFrame([sorted(f) for f in families])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [],
   "source": [
    "fam.sort_values(by=fam.columns.tolist(), inplace=True)\n",
    "fam.sort_values(by=fam.index.tolist(), axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HAIN001_00088 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN002_00089 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN003_00595 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN004_00583 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN005_00080 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN006_00083 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN007_00079 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN008_00566 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN009_00619 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN010_00854 Cystathionine gamma-synthase/O-a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HAIN001_00093 Threonine synthase</td>\n",
       "      <td>HAIN002_00162 Threonine synthase</td>\n",
       "      <td>HAIN003_00590 Threonine synthase</td>\n",
       "      <td>HAIN004_00589 Threonine synthase</td>\n",
       "      <td>HAIN005_00130 Threonine synthase</td>\n",
       "      <td>HAIN006_00088 Threonine synthase</td>\n",
       "      <td>HAIN007_00151 Threonine synthase</td>\n",
       "      <td>HAIN008_00561 Threonine synthase</td>\n",
       "      <td>HAIN009_00513 Threonine synthase</td>\n",
       "      <td>HAIN010_00849 Threonine synthase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>HAIN002_00093 hypothetical protein</td>\n",
       "      <td>HAIN007_00083 hypothetical protein</td>\n",
       "      <td>HAIN009_00615 hypothetical protein</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HAIN002_00094 replicative DNA helicase</td>\n",
       "      <td>HAIN007_00084 replicative DNA helicase</td>\n",
       "      <td>HAIN009_00614 replicative DNA helicase</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>HAIN002_00095 hypothetical protein</td>\n",
       "      <td>HAIN007_00085 hypothetical protein</td>\n",
       "      <td>HAIN009_00613 hypothetical protein</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     0  \\\n",
       "30   HAIN001_00088 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN001_00093 Threonine synthase   \n",
       "100                 HAIN002_00093 hypothetical protein   \n",
       "20              HAIN002_00094 replicative DNA helicase   \n",
       "60                  HAIN002_00095 hypothetical protein   \n",
       "\n",
       "                                                     1  \\\n",
       "30   HAIN002_00089 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN002_00162 Threonine synthase   \n",
       "100                 HAIN007_00083 hypothetical protein   \n",
       "20              HAIN007_00084 replicative DNA helicase   \n",
       "60                  HAIN007_00085 hypothetical protein   \n",
       "\n",
       "                                                     2  \\\n",
       "30   HAIN003_00595 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN003_00590 Threonine synthase   \n",
       "100                 HAIN009_00615 hypothetical protein   \n",
       "20              HAIN009_00614 replicative DNA helicase   \n",
       "60                  HAIN009_00613 hypothetical protein   \n",
       "\n",
       "                                                     3  \\\n",
       "30   HAIN004_00583 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN004_00589 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                                     4  \\\n",
       "30   HAIN005_00080 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN005_00130 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                                     5  \\\n",
       "30   HAIN006_00083 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN006_00088 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                                     6  \\\n",
       "30   HAIN007_00079 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN007_00151 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                                     7  \\\n",
       "30   HAIN008_00566 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN008_00561 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                                     8  \\\n",
       "30   HAIN009_00619 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN009_00513 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                                     9  \n",
       "30   HAIN010_00854 Cystathionine gamma-synthase/O-a...  \n",
       "19                    HAIN010_00849 Threonine synthase  \n",
       "100                                               None  \n",
       "20                                                None  \n",
       "60                                                None  "
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The next cells are extra code to correct a major bug\n",
    "fam2 = fam  # Copy the panda dataframe\n",
    "\n",
    "# Adjust the column names by simply assign it to the .columns attribute:\n",
    "fam2.columns = strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HAIN001</th>\n",
       "      <th>HAIN002</th>\n",
       "      <th>HAIN003</th>\n",
       "      <th>HAIN004</th>\n",
       "      <th>HAIN005</th>\n",
       "      <th>HAIN006</th>\n",
       "      <th>HAIN007</th>\n",
       "      <th>HAIN008</th>\n",
       "      <th>HAIN009</th>\n",
       "      <th>HAIN010</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HAIN001_00088 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN002_00089 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN003_00595 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN004_00583 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN005_00080 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN006_00083 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN007_00079 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN008_00566 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN009_00619 Cystathionine gamma-synthase/O-a...</td>\n",
       "      <td>HAIN010_00854 Cystathionine gamma-synthase/O-a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HAIN001_00093 Threonine synthase</td>\n",
       "      <td>HAIN002_00162 Threonine synthase</td>\n",
       "      <td>HAIN003_00590 Threonine synthase</td>\n",
       "      <td>HAIN004_00589 Threonine synthase</td>\n",
       "      <td>HAIN005_00130 Threonine synthase</td>\n",
       "      <td>HAIN006_00088 Threonine synthase</td>\n",
       "      <td>HAIN007_00151 Threonine synthase</td>\n",
       "      <td>HAIN008_00561 Threonine synthase</td>\n",
       "      <td>HAIN009_00513 Threonine synthase</td>\n",
       "      <td>HAIN010_00849 Threonine synthase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>HAIN002_00093 hypothetical protein</td>\n",
       "      <td>HAIN007_00083 hypothetical protein</td>\n",
       "      <td>HAIN009_00615 hypothetical protein</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HAIN002_00094 replicative DNA helicase</td>\n",
       "      <td>HAIN007_00084 replicative DNA helicase</td>\n",
       "      <td>HAIN009_00614 replicative DNA helicase</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>HAIN002_00095 hypothetical protein</td>\n",
       "      <td>HAIN007_00085 hypothetical protein</td>\n",
       "      <td>HAIN009_00613 hypothetical protein</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               HAIN001  \\\n",
       "30   HAIN001_00088 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN001_00093 Threonine synthase   \n",
       "100                 HAIN002_00093 hypothetical protein   \n",
       "20              HAIN002_00094 replicative DNA helicase   \n",
       "60                  HAIN002_00095 hypothetical protein   \n",
       "\n",
       "                                               HAIN002  \\\n",
       "30   HAIN002_00089 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN002_00162 Threonine synthase   \n",
       "100                 HAIN007_00083 hypothetical protein   \n",
       "20              HAIN007_00084 replicative DNA helicase   \n",
       "60                  HAIN007_00085 hypothetical protein   \n",
       "\n",
       "                                               HAIN003  \\\n",
       "30   HAIN003_00595 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN003_00590 Threonine synthase   \n",
       "100                 HAIN009_00615 hypothetical protein   \n",
       "20              HAIN009_00614 replicative DNA helicase   \n",
       "60                  HAIN009_00613 hypothetical protein   \n",
       "\n",
       "                                               HAIN004  \\\n",
       "30   HAIN004_00583 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN004_00589 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                               HAIN005  \\\n",
       "30   HAIN005_00080 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN005_00130 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                               HAIN006  \\\n",
       "30   HAIN006_00083 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN006_00088 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                               HAIN007  \\\n",
       "30   HAIN007_00079 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN007_00151 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                               HAIN008  \\\n",
       "30   HAIN008_00566 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN008_00561 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                               HAIN009  \\\n",
       "30   HAIN009_00619 Cystathionine gamma-synthase/O-a...   \n",
       "19                    HAIN009_00513 Threonine synthase   \n",
       "100                                               None   \n",
       "20                                                None   \n",
       "60                                                None   \n",
       "\n",
       "                                               HAIN010  \n",
       "30   HAIN010_00854 Cystathionine gamma-synthase/O-a...  \n",
       "19                    HAIN010_00849 Threonine synthase  \n",
       "100                                               None  \n",
       "20                                                None  \n",
       "60                                                None  "
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Note the Families names, this info should disappear\n",
    "fam2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Voici la commande qui corrige le bug\n",
    "# On itere sur le dataframe (par colonne) et on remplace une expression regulieres dans chacune des rangees\n",
    "for i in strains:\n",
    "    fam2[i]= fam2[i].str.replace(r'(\\S+_\\d+)(.+)', r\"\\1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HAIN001</th>\n",
       "      <th>HAIN002</th>\n",
       "      <th>HAIN003</th>\n",
       "      <th>HAIN004</th>\n",
       "      <th>HAIN005</th>\n",
       "      <th>HAIN006</th>\n",
       "      <th>HAIN007</th>\n",
       "      <th>HAIN008</th>\n",
       "      <th>HAIN009</th>\n",
       "      <th>HAIN010</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HAIN001_00088</td>\n",
       "      <td>HAIN002_00089</td>\n",
       "      <td>HAIN003_00595</td>\n",
       "      <td>HAIN004_00583</td>\n",
       "      <td>HAIN005_00080</td>\n",
       "      <td>HAIN006_00083</td>\n",
       "      <td>HAIN007_00079</td>\n",
       "      <td>HAIN008_00566</td>\n",
       "      <td>HAIN009_00619</td>\n",
       "      <td>HAIN010_00854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HAIN001_00093</td>\n",
       "      <td>HAIN002_00162</td>\n",
       "      <td>HAIN003_00590</td>\n",
       "      <td>HAIN004_00589</td>\n",
       "      <td>HAIN005_00130</td>\n",
       "      <td>HAIN006_00088</td>\n",
       "      <td>HAIN007_00151</td>\n",
       "      <td>HAIN008_00561</td>\n",
       "      <td>HAIN009_00513</td>\n",
       "      <td>HAIN010_00849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>HAIN002_00093</td>\n",
       "      <td>HAIN007_00083</td>\n",
       "      <td>HAIN009_00615</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HAIN002_00094</td>\n",
       "      <td>HAIN007_00084</td>\n",
       "      <td>HAIN009_00614</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>HAIN002_00095</td>\n",
       "      <td>HAIN007_00085</td>\n",
       "      <td>HAIN009_00613</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           HAIN001        HAIN002        HAIN003        HAIN004  \\\n",
       "30   HAIN001_00088  HAIN002_00089  HAIN003_00595  HAIN004_00583   \n",
       "19   HAIN001_00093  HAIN002_00162  HAIN003_00590  HAIN004_00589   \n",
       "100  HAIN002_00093  HAIN007_00083  HAIN009_00615           None   \n",
       "20   HAIN002_00094  HAIN007_00084  HAIN009_00614           None   \n",
       "60   HAIN002_00095  HAIN007_00085  HAIN009_00613           None   \n",
       "\n",
       "           HAIN005        HAIN006        HAIN007        HAIN008  \\\n",
       "30   HAIN005_00080  HAIN006_00083  HAIN007_00079  HAIN008_00566   \n",
       "19   HAIN005_00130  HAIN006_00088  HAIN007_00151  HAIN008_00561   \n",
       "100           None           None           None           None   \n",
       "20            None           None           None           None   \n",
       "60            None           None           None           None   \n",
       "\n",
       "           HAIN009        HAIN010  \n",
       "30   HAIN009_00619  HAIN010_00854  \n",
       "19   HAIN009_00513  HAIN010_00849  \n",
       "100           None           None  \n",
       "20            None           None  \n",
       "60            None           None  "
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam2.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Format data to plot \n",
    "\n",
    "We use and old script I wrote which is quite ugly to read, but works. You can find it here: [spot_ICE](https://gitlab.pasteur.fr/gem/spot_ICE)\n",
    "\n",
    "The advantage is that it produces an interactive plot ot the spot, but the drawback is that the input format is a bit weird. Let's get through\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqUtils"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The input file of the script looks like:\n",
    "\n",
    "    #header\n",
    "    \\n\n",
    "    ICE_ID\n",
    "    genome1&2 genome2&2 genome3&50 genome4&2 #list of genome ID with the number of gene in between the 2 core genes (included)\n",
    "    <Table with genes from the genomes with the ICE in columns, see below for description>\n",
    "    0.4 0.45 0.43 .... #GC% values along the spot.\n",
    "    \\n\n",
    "Where the table has the following columns:\n",
    "\n",
    "|Column|description|Value| Comment|\n",
    "|-:|:-|:-|:-:|\n",
    "|1| Gene_id | String ||\n",
    "|2| Annotation| String ||\n",
    "|3| Start| Integer ||\n",
    "|4| End| Integer||\n",
    "|5| Gene orientation | 1/- 1||\n",
    "|6| Replicon orientation | 1/-1 | compare to the other genome of the species, is this replicon sequenced on the same strand or not ? (same average orientation of the core genes)|\n",
    "|7| Normalized orientation | (Gene orientation * Replicon orientation)||\n",
    "|8| Gene size| Integer||\n",
    "|9| MPF profile | MOBH/G_tfc23/...| if the gene is matched by macsyfinder, which profile matched ?|\n",
    "|10| Integrase_type | Tyr_rec | if the gene is matched by tyrosine or serine recombinase profiles|\n",
    "|11| Gene families | HAIN002_00160&HAIN009_00555 | genes in other genome of the species that are homologous.|\n",
    "|12| RNA |tRNA&tRNA-Lys&143658&143734&1 | RFAM hits, in the line of the closest gene, and formatted as: RNA_type&RFAM_number&start&end&strand|\n",
    "\n",
    "In any case, if there is not value, it must be a 0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n"
     ]
    }
   ],
   "source": [
    "# Slightly edited by JSB to ensure that gene_fams infos look like:\n",
    "        # HAIN002_00089    HAIN001_00088&HAIN003_00595&HAIN004_00583&HAIN...\n",
    "        # HAIN002_00093                          HAIN007_00083&HAIN009_00615\n",
    "        # HAIN002_00094                          HAIN007_00084&HAIN009_00614\n",
    "        # HAIN002_00095                          HAIN007_00085&HAIN009_00613\n",
    "        # HAIN002_00096                          HAIN007_00086&HAIN009_00612\n",
    "\n",
    "# The code below use fam2 panda dataframe rather than fam           \n",
    "\n",
    "with open(\"All_data_hain.txt\", \"w\") as outf:\n",
    "    outf.write(\"#Annotation\\tgene_name\\tStart\\tEnd\\tStrand\\tStrand_replicon\\tStrand_norm\\tSize\\tGene\\tIntegrase\\tgene_fam\\tRNA\\n\\n\") # Header optional, could be an empty line\n",
    "\n",
    "    for sys_ID in conj.System_Id.unique():\n",
    "        df_sys_ID = pd.DataFrame(columns=fam2.columns[:-1])\n",
    "        strain = sys_ID.split(\"_\")[0]\n",
    "        for f in fam2.iterrows():\n",
    "            try:\n",
    "                idx = f[1][f[1].str.contains(strain, na=False)].index\n",
    "                df_sys_ID.loc[f[1][idx[0]]] = f[1][f[1].index.isin(idx)==0].values\n",
    "            except IndexError:\n",
    "                pass\n",
    "        df_sys_ID.sort_index(inplace=True)\n",
    "        # spot_size is like: ['HAIN001&2', 'HAIN002&70', 'HAIN003&2', 'HAIN004&3', 'HAIN005&46', ...]\n",
    "        spot_size = [\"&\".join((m, str(n))) for m,n in pd.Series([i.split(\"_\")[0].strip() for i in fam2.values.ravel() if isinstance(i, str)]).value_counts().sort_index().iteritems()]\n",
    "        gbk = SeqIO.read(\"Sequences/Annotations/{strain}/{strain}.gbk\".format(strain=strain), \"genbank\")\n",
    "\n",
    "        final = pd.DataFrame(columns=[\"Annotation\", \"gene_name\", \"Start\", \"End\", \"Strand\"])\n",
    "        for feat in gbk.features[1:]:\n",
    "            if min(flanking_core_ID[strain]) <= feat.qualifiers[\"locus_tag\"][0] <= max(flanking_core_ID[strain]):\n",
    "                if \"gene\" in feat.qualifiers:\n",
    "                    gene_name = feat.qualifiers[\"gene\"][0]\n",
    "                elif feat.type==\"tRNA\":\n",
    "                    gene_name = feat.qualifiers[\"note\"][0]\n",
    "                else:\n",
    "                    gene_name = \"HP\"\n",
    "                final.loc[feat.qualifiers[\"locus_tag\"][0]] = feat.qualifiers[\"product\"][0], gene_name, feat.location.start, feat.location.end, feat.location.strand\n",
    "\n",
    "        final[\"Strand_replicon\"] = sequencing_orientation[strain]\n",
    "        final[\"Strand_norm\"] = final.Strand*final.Strand_replicon\n",
    "        final[\"Size\"] = final.End - final.Start + 1\n",
    "        final = final.merge(conj.set_index(\"#Hit_Id\")[[\"Gene\"]], left_index=True, right_index=True, how=\"left\")\n",
    "        final[\"Integrase\"] = np.nan # If you detected integrases you can put something else than Nan or 0 in front of the corresponding gene.\n",
    "        gene_fams = df_sys_ID.apply(lambda x: \"&\".join(x.dropna()), axis=1).replace(\"\", \"0\")\n",
    "        \n",
    "        # gene_fams is like:\n",
    "        # HAIN002_00089    HAIN001_00088&HAIN003_00595&HAIN004_00583&HAIN...\n",
    "        # HAIN002_00093                          HAIN007_00083&HAIN009_00615\n",
    "        # HAIN002_00094                          HAIN007_00084&HAIN009_00614\n",
    "        # HAIN002_00095                          HAIN007_00085&HAIN009_00613\n",
    "        # HAIN002_00096                          HAIN007_00086&HAIN009_00612\n",
    "             \n",
    "        final = final.merge(gene_fams.to_frame(name=\"gene_fam\"), left_index=True, right_index=True, how='left')\n",
    "        final[\"RNA\"] = final.apply(lambda x: \"&\".join((\"tRNA\",\n",
    "                                                       x[\"Annotation\"],\n",
    "                                                       str(int(x[\"Start\"])),\n",
    "                                                       str(int(x[\"End\"])),\n",
    "                                                       str(int(x[\"Strand_norm\"])))) if \"tRNA\" in x[\"Annotation\"] else np.nan, axis=1)\n",
    "\n",
    "        final.fillna(0, inplace=True)\n",
    "        final[final.select_dtypes([float]).columns] = final.select_dtypes([float]).astype(int, inplace=1) \n",
    "        GC = [SeqUtils.GC(gbk.seq[final.Start[0]:final.End[-1]][i:i+1000]) for i in range(0, final.End[-1]-final.Start[0]+1, 1000)]\n",
    "        outf.write(sys_ID+\"\\n\")\n",
    "        outf.write(\" \".join(spot_size)+\"\\n\")\n",
    "        final.to_csv(outf, header=None, sep=\"\\t\")\n",
    "        outf.write(\" \".join(map(str, GC))+\"\\n\\n\")\n",
    "print \"Done\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plot the spot\n",
    "\n",
    "Once you downloaded the script provided earlier, just type (in a terminal preferably, and ***not*** in a interactive python environnenment (note the *!* at the beginning of the line, call a shell)\n",
    "\n",
    "The window can be clicked:\n",
    "\n",
    "- left click on gene: print information for that genes\n",
    "- left click on any x-axis: erase all gene informations\n",
    "- right click on 2 genes in the same ICE: select the gene as a first and last genes delimiting the ICE\n",
    "- shift + right-click: remove the selected genes in the corresponding ICE.\n",
    "    - Beware to click 2 times in the same ICE otherwise the limits will be mixed between ICEs (in case of odd number of right-clicks).\n",
    "    - If you click 4 times in the same ICE, it will define 2 ICEs in tandem. (Or any even number of right click will define half as many ICE)\n",
    "- The spot are plotted 4 at a times, and the window close every 4. You can increase this number in the script if you have a big enough screen. \n",
    "- You can export an ICE in its spot that you'd like by ticking the box on the right (weird rectangles) and click export. It will create a file `Selection...` that you can re-use. You could easily merge 2 exported files from different window. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rouler le script dans un terminal ordinaire \n",
    "#./plot_ICE_spot.py All_data_hain.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here is a screenshot, obtained with this entire pipeline:\n",
    "<img src=\"Screenshot_spot_ICE.png\" alt=\"spot_ICE\" style=\"width: 75%;\"/>\n"
   ]
  }
 ],
 "metadata": {
  "hide_input": false,
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  },
  "toc": {
   "toc_cell": false,
   "toc_number_sections": true,
   "toc_section_display": "block",
   "toc_threshold": 6,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
